### DISCUSSION TITLE
Github Pages meta CSP

### DISCUSSION THREAD

**Discussion by bnjmnrsh** [KindSpells/astro-shield#85] [Q&A]
Hi @castarco thank you for taking this project on. 
I see that you have issues created for creating CSP headers on platforms like [Cloudflare](https://github.com/kindspells/astro-shield/issues/63), [Vercel](https://github.com/kindspells/astro-shield/issues/65) and [Netlify](https://github.com/kindspells/astro-shield/issues/65). Github Pages, however [does not provide an API for modifying headers](https://github.com/orgs/community/discussions/54257), and so setting a meta tag with a CSP header is the best we can do in this environment. 

Do you have any recommended approaches for using astro-sheild on Github Pages? For example, would it be possible or advisable to make  `getGlobalHashes` available in order to set these as values in a CSP manually defined in a meta tag?

**Comment by castarco** [comment]
Hi @bnjmnrsh , I'm not sure it is possible to easily configure headers in Github Pages... 🤔 , I actually never tried, but I would be pleasantly surprised if that was possible. The few times I checked in the past I didn't find anything.

Regarding meta tags... that's an interesting idea. Not sure it's great, but beats doing nothing. Feel free to open an issue describing how you envision such a feature! 😄 .
